includes/orders/functions/orders.php

```diff
@@ -1049,16 +1049,23 @@ function edd_build_order( $order_data = array() ) {
 			$discount = edd_get_discount_by( 'code', $discount );
 
 			if ( $discount ) {
-				$discounted_amount = $subtotal - $discount->get_discounted_amount( $subtotal );
+				$discount_amount = 0;
+				$items           = $order_data['cart_details'];
+
+				if ( is_array( $items ) && ! empty( $items ) ) {
+					foreach ( $items as $key => $item ) {
+						$discount_amount += edd_get_item_discount_amount( $item, $items, array( $discount ) );
+					}
+				}
 
 				edd_add_order_adjustment( array(
 					'object_id'   => $order_id,
 					'object_type' => 'order',
```

Comment:
Since you have worked out how to calculate the discount per item, what about storing the order adjustment per `order_item` directly instead of per `order`? This seems like it could make retrieving the discount amounts easier when looking at refunds, and would be consistent with how we store `tax_rate` on orders.

Reply:
@robincornett I know you've done some more research regarding Adjustments since this comment -- do you still feel the same way?

I think it could be done separately as this implementation is meant to mimic the current frontend checkout.

Reply:
per @cklosowski, Discounts should [stay assigned](https://github.com/easydigitaldownloads/easy-digital-downloads/pull/8014#issuecomment-655160388) to `order`s, so I think this can stay as is:
> As far as discounts, this gets a little difficult. The discount code is applied to the order as a whole. This is ideal as it allows for accurate reporting of how often a discount is used. As far as it affecting each item in the cart, that's likely because discounts can be cart wide or specific to a single item (based off the discount code settings). As EDD has always done, a discount code being attached to an order simply adjusts the order total, by adjusting each order item, so that items that can be affected are and others that are not affected by the discount code can be skipped. This also allows for accurate reporting of earnings of a product as well as discount code usages from a reporting standpoint.

In the comment after that one, I added a link to a spreadsheet I created with a record of every use of `edd_add_order_adjustment`, trying to evaluate purely at a data level, along with some notes/questions. If you have some time to look over the GH comments and weigh in, I would appreciate it, @spencerfinnell.